Source/WebCore/page/Quirks.cpp

```diff
@@ -1901,4 +1901,19 @@ std::optional<TargetedElementSelectors> Quirks::defaultVisibilityAdjustmentSelec
 #endif
 }
 
+String Quirks::scriptToEvaluateBeforeRunningScriptFromURL(const URL& scriptURL)
```

Comment:
Neat!

Reply:
There is an opportunity for a generic function: `overrideWindowObject('property_name', object)`

Thanks for reducing the scope this is cool. It goes into what we want to do with @stwrt